src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/CancelInvocationMessage.java

```diff
@@ -3,14 +3,32 @@
 
 package com.microsoft.signalr;
 
+import java.util.Map;
+
 final class CancelInvocationMessage extends HubMessage {
     private final int type = HubMessageType.CANCEL_INVOCATION.value;
+    private Map<String, String> headers;
     private final String invocationId;
 
     public CancelInvocationMessage(String invocationId) {
+        this(null, invocationId);
+    }
+    
+    public CancelInvocationMessage(Map<String, String> headers, String invocationId) {
```

Comment:
These HubMessages are all package-private, right? Should we just change the ctor instead of adding an overload?

Reply:
Yeah, I could just give each Message one ctor & modify the uses of them throughout

Reply:
> These HubMessages are all package-private, right?

It's worth noting we probably want MessagePack to be in a new package

Reply:
Updated so each message type has one ctor